static/js/components/Drawer.js

```diff
@@ -13,21 +14,26 @@ import EditProfileDisplay from "./EditProfileDisplay"
 import ViewProfileDisplay from "./ViewProfileDisplay"
 import PaymentDisplay from "./PaymentDisplay"
 
-const renderDrawerContents = (drawerState: string): ?React$Element<*> => {
+const renderDrawerContents = (
+  drawerState: string,
+  drawerMeta?: Object
+): ?React$Element<*> => {
+  let props
   switch (drawerState) {
   case PROFILE_EDIT:
     return <EditProfileDisplay />
   case PROFILE_VIEW:
     return <ViewProfileDisplay />
   case PAYMENT:
-    return <PaymentDisplay />
+    props = { application: R.prop("application", drawerMeta) }
+    return <PaymentDisplay {...props} />
```

Comment:
Can simplify this down to:
```suggestion
    return <PaymentDisplay application={R.prop("application", drawerMeta)} />
```